### PR TITLE
[fix] Various fixes for TF AWS 3.0 compatibility

### PR DESCRIPTION
**Pull request by mbarrien**
* Shorten static site lambda name
* Remove unneeded bare aws provider from static site
* Fix aws-ssm-params readme
* Upgrade terraform-aws-modules/security-group/aws for TF 3.0 compatiblity (and update since new version renames output this_security_group_id to security_group_id)
* aws-lambda-functionaws_cloudwatch_log_group use ARN instead of manually
      constructing since ARN is fixed in TF AWS 3.0
* Update readmes via make docs

### CODE REVIEW

File: aws-ecs-service/alb.tf [hunk 1]
```diff
@@ -30,7 +30,7 @@ resource "aws_lb_target_group" "service" {
 resource "aws_lb" "service" {
```
Comment (cziie) [inline aws-ecs-service/alb.tf:30]: ⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing version constraint for provider "aws" in "required_providers"

File: aws-redis-node/main.tf [hunk 2]
```diff
@@ -56,7 +56,7 @@ resource "aws_elasticache_cluster" "default" {
   num_cache_nodes      = 1
   parameter_group_name = var.parameter_group_name
```
Comment (cziie) [inline aws-redis-node/main.tf:57]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>"default.redis5.0" is default parameter group. You cannot edit it.

File: aws-single-page-static-site/main.tf [hunk 3]
```diff
@@ -87,7 +87,7 @@ resource "aws_s3_bucket_public_access_block" "bucket" {
 module "security_headers_lambda" {
   source = "../aws-lambda-edge-add-security-headers"
 
-  function_name = replace("${local.website_fqdn}-static-site-security-headers", ".", "-")
+  function_name = replace("${local.website_fqdn}-headers", ".", "-")
```
Comment (mbarrien) [inline aws-single-page-static-site/main.tf:90]: Reduce issue where this string is >64 characters long. Forces redeploy of lambda for anything already deployed with HSTS headers (which should just be a single deployment in Along), and the redeploy should be safe.

File: aws-lambda-function/main.tf [hunk 4]
```diff
@@ -81,14 +81,6 @@ data "aws_region" "current" {}
 data "aws_caller_identity" "current" {}
```
Comment (cziie) [inline aws-lambda-function/main.tf:81]: ⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>data "aws_caller_identity" "current" is declared but not used

**Review by alexsmaliy-czi — APPROVED** [review]

**Pull request merged**